backend/services/workflows/app/processor/job.go

```diff
@@ -61,6 +63,18 @@ func (j JobProcessor) ProcessJSON(
 				len(key) > len(workflowInputVariable) {
 				key = key[len(workflowInputVariable):]
 				for _, param := range j.job.InputParameters {
+					// support for key === "device.json.tier"; accessing the input params
+					// objects by json identifiers. for instance if a whole device is passed
+					// as an input parameter then with `device.json.tier` you can refer
+					// to the Device.Tier field
+					if strings.HasPrefix(key, param.Name+".json") {
+						return gojsonq.New().FromString(param.Value).Find(
+							strings.TrimPrefix(
+								key,
+								param.Name+".json.",
+							),
+						)
+					}
```

Comment:
Shouldn't this be in the `ProcessJobString` interface so it applies uniformly to all kinds of strings?